ios/RNSScreen.mm

```diff
@@ -391,72 +599,183 @@ @implementation RNSScreen {
+#ifdef RN_FABRIC_ENABLED
+  [_initialView notifyAppear];
+#else
+  if (!_isSwiping || _shouldNotify) {
+    // we are going forward or dismissing without swipe
+    // or successfully swiped back
+    [((RNSScreenView *)self.view) notifyAppear];
+    [self notifyTransitionProgress:1.0 closing:NO goingForward:_goingForward];
   }
-  return UIInterfaceOrientationMaskAllButUpsideDown;
+
+  _isSwiping = NO;
+  _shouldNotify = YES;
+#endif
 }
 
-- (UIViewController *)childViewControllerForHomeIndicatorAutoHidden
+- (void)viewDidDisappear:(BOOL)animated
 {
-  UIViewController *vc = [self findChildVCForConfigAndTrait:RNSWindowTraitHomeIndicatorHidden includingModals:YES];
-  return vc == self ? nil : vc;
+  [super viewDidDisappear:animated];
+#ifdef RN_FABRIC_ENABLED
+  [_initialView notifyDisappear];
+  if (self.parentViewController == nil && self.presentingViewController == nil) {
+    // screen dismissed, send event
+    [_initialView notifyDismissedWithCount:1];
+  }
+#else
+  if (self.parentViewController == nil && self.presentingViewController == nil) {
+    if (((RNSScreenView *)self.view).preventNativeDismiss) {
+      // if we want to prevent the native dismiss, we do not send dismissal event,
+      // but instead call `updateContainer`, which restores the JS navigation stack
+      [((RNSScreenView *)self.view).reactSuperview updateContainer];
+      [((RNSScreenView *)self.view) notifyDismissCancelledWithDismissCount:_dismissCount];
+    } else {
+      // screen dismissed, send event
+      [((RNSScreenView *)self.view) notifyDismissedWithCount:_dismissCount];
+    }
+  }
+
+  // same flow as in viewDidAppear
+  if (!_isSwiping || _shouldNotify) {
+    [((RNSScreenView *)self.view) notifyDisappear];
+    [self notifyTransitionProgress:1.0 closing:YES goingForward:_goingForward];
+  }
+
+  _isSwiping = NO;
+  _shouldNotify = YES;
+
+  [self traverseForScrollView:self.view];
+#endif
 }
 
-- (BOOL)prefersHomeIndicatorAutoHidden
+- (void)viewDidLayoutSubviews
 {
-  return ((RNSScreenView *)self.view).homeIndicatorHidden;
+  [super viewDidLayoutSubviews];
+#ifdef RN_FABRIC_ENABLED
+  BOOL isDisplayedWithinUINavController = [self.parentViewController isKindOfClass:[UINavigationController class]];
+  if (isDisplayedWithinUINavController) {
+    [_initialView updateBounds];
+  }
+#else
+  // The below code makes the screen view adapt dimensions provided by the system. We take these
+  // into account only when the view is mounted under RNScreensNavigationController in which case system
+  // provides additional padding to account for possible header, and in the case when screen is
+  // shown as a native modal, as the final dimensions of the modal on iOS 12+ are shorter than the
+  // screen size
+  BOOL isDisplayedWithinUINavController =
+      [self.parentViewController isKindOfClass:[RNScreensNavigationController class]];
+  BOOL isPresentedAsNativeModal = self.parentViewController == nil && self.presentingViewController != nil;
```

Comment:
We should check if `isPresentedAsNativeModal` should not also be needed for both architectures to provide correct frames in modal. If so, I think we can extract those 2 BOOLs outside `ifdef` since they are the same (`UINavigationController` should be changed to `RNScreensNavigationController`)

Reply:
As we discussed, I noted this down and will address it in following PR

Reply:
20a0a92